azurerm/internal/services/devtestlabs/resource_arm_dev_test_global_shutdown_schedule.go

```diff
@@ -0,0 +1,324 @@
+	rid, err := azure.ParseAzureResourceID(targetResourceID)
+	if err != nil {
+		return err
+	}
+
+	vmName := rid.Path["virtualMachines"]
+	resGroup := rid.ResourceGroup
+
+	// Can't find any official documentation on this, but the API returns a 400 for any other name.
+	// The best example I could find is here: https://social.msdn.microsoft.com/Forums/en-US/25a02403-dba9-4bcb-bdcc-1f4afcba5b65/powershell-script-to-autoshutdown-azure-virtual-machine?forum=WAVirtualMachinesforWindows
+	name := "shutdown-computevm-" + vmName
+
+	if features.ShouldResourcesBeImported() && d.IsNewResource() {
+		existing, err := client.Get(ctx, resGroup, name, "")
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of existing Schedule %q (Resource Group %q): %s", name, resGroup, err)
+			}
+		}
+
+		if existing.ID != nil && *existing.ID != "" {
+			return tf.ImportAsExistsError("azurerm_dev_test_global_shutdown_schedule", *existing.ID)
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	taskType := "ComputeVmShutdownTask"
+	t := d.Get("tags").(map[string]interface{})
+
+	schedule := dtl.Schedule{
+		Location: &location,
+		ScheduleProperties: &dtl.ScheduleProperties{
+			TargetResourceID: &targetResourceID,
+			TaskType:         &taskType,
+		},
+		Tags: tags.Expand(t),
+	}
+
+	switch status := d.Get("status"); status {
+	case string(dtl.EnableStatusEnabled):
+		schedule.ScheduleProperties.Status = dtl.EnableStatusEnabled
+	case string(dtl.EnableStatusDisabled):
+		schedule.ScheduleProperties.Status = dtl.EnableStatusDisabled
+	default:
+	}
+
+	if timeZoneId := d.Get("time_zone_id").(string); timeZoneId != "" {
+		schedule.ScheduleProperties.TimeZoneID = &timeZoneId
+	}
+
+	if v, ok := d.GetOk("daily_recurrence"); ok {
+		dailyRecurrence := expandArmDevTestLabGlobalShutdownScheduleRecurrenceDaily(v)
+		schedule.DailyRecurrence = dailyRecurrence
+	}
+
+	if _, ok := d.GetOk("notification_settings"); ok {
+		notificationSettings := expandArmDevTestLabGlobalShutdownScheduleNotificationSettings(d)
+		schedule.NotificationSettings = notificationSettings
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resGroup, name, schedule); err != nil {
+		return err
+	}
+
+	read, err := client.Get(ctx, resGroup, name, "")
+	if err != nil {
+		return err
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read Dev Test Global Schedule %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmDevTestLabGlobalShutdownScheduleRead(d, meta)
+}
+
+func resourceArmDevTestLabGlobalShutdownScheduleRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DevTestLabs.GlobalLabSchedulesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["schedules"]
```

Comment:
Could we write a parse ID function like seen in #5779 ?

Reply:
Added a function for this as suggested.